src/pf-modal/pf-modal.component.js

```diff
@@ -0,0 +1,250 @@
+   */
+  attributeChangedCallback (attrName, oldValue, newValue) {
+    // When the modal is shown, update keyboard/screen reader behavior.
+    if (attrName === 'open' && newValue !== null) {
+      this._showModal();
+    }
+    if (attrName === 'keyboard') {
+      if (newValue !== null) {
+        this._element.addEventListener('keydown', this._keydownHandler, false);
+      } else {
+        this._element.removeEventListener('keydown', this._keydownHandler, false);
+      }
+    }
+  }
+
+  /*
+   * Called when an instance of the element is created
+   * We build up the complete DOM tree of web components
+   */
+  createdCallback () {
+    this._element = this;
+    // Decorate the pf-modal with bootstrap modal style
+    this._element.className = 'modal fade';
+    this._element.setAttribute('tabindex', -1);
+    let headerPart = this._element.querySelector('[modalPart="header"]').innerHTML;
+    let bodyPart = this._element.querySelector('[modalPart="body"]').innerHTML;
+    let footerPart = this._element.querySelector('[modalPart="footer"]').innerHTML;
+    this._element.innerHTML = tmpl;
+    this._element.querySelector('.modal-header').innerHTML = headerPart;
+    this._element.querySelector('.modal-body').innerHTML = bodyPart;
+    this._element.querySelector('.modal-footer').innerHTML = footerPart;
+
+    this._target = document.querySelector(this.getAttribute('targetSelector'));
+    this._dialog = this._element.querySelector('.modal-dialog');
+    this._mask = null;
+  }
+
+  /*
+   * Called when an instance was inserted into the document
+   * So we place event bindings here
+   */
+  attachedCallback () {
+    if (this.open) {
+      this._showModal();
+    }
+    this._target.addEventListener('click', this.show.bind(this), false);
+    this._element.addEventListener('click', this.hide, false);
+    if (this.keyboard) {
+      this._element.addEventListener('keydown', this._keydownHandler, false);
+    }
+  }
+
+  /*
+   * Show the modal when attribute open is added or property open is enabled
+   * @private
+   */
+  _showModal () {
+    pfUtil.addClass(document.body, 'modal-open');
+    this._mask = document.createElement('div');
+    this._mask.className = 'modal-backdrop fade';
+    document.body.appendChild(this._mask);
+    pfUtil.reflow(this._mask);
+    pfUtil.addClass(this._mask, 'in');
+    pfUtil.reflow(this._element);
+    pfUtil.addClass(this._element, 'in');
+    pfUtil.once(this._dialog, 'transitionend', this._afterShowModal, this);
+  }
+
+  /*
+   * Show the modal
+   * @public
+   */
+  show () {
+    this._element.dispatchEvent(new CustomEvent('show.bs.modal', { 'detail': { 'relatedTarget': this._target }}));
+    if (this.open) {
+      return;
+    }
+    this.open = true;
+  }
+
+  /*
+   * Callback after modal is shown
+   * @private
+   */
+  _afterShowModal () {
+    this._element.focus();
+    this._element.dispatchEvent(new CustomEvent('shown.bs.modal', { 'detail': { 'relatedTarget': this._target }}));
```

Comment:
We should document exactly what events, names, and properties are generated. If not for this private function, at least for the public ones. And perhaps also at the top of the component's JsDoc to make it clear what public events are available?